vividus-to-zephyr-exporter/src/main/java/org/vividus/zephyr/ZephyrFacade.java

```diff
@@ -0,0 +1,130 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.zephyr;
+
+import static org.apache.commons.lang3.Validate.isTrue;
+import static org.apache.commons.lang3.Validate.notBlank;
+import static org.apache.commons.lang3.Validate.notEmpty;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.vividus.jira.JiraClient;
+import org.vividus.jira.JiraFacade;
+import org.vividus.jira.model.Project;
+import org.vividus.jira.model.Version;
+import org.vividus.util.json.JsonPathUtils;
+
+public class ZephyrFacade implements IZephyrFacade
+{
+    private static final String ZAPI_ENDPOINT = "/rest/zapi/latest/";
+    private static final String CREATE_EXECUTION_ENDPOINT = ZAPI_ENDPOINT + "execution/";
+    private static final String UPDATE_EXECUTION_STATUS_ENDPOINT = ZAPI_ENDPOINT + "execution/%s/execute";
+
+    private static final String FOLDER_ID_JSON_PATH = "$.[?(@.folderName=='%s')].folderId";
+    private static final String EXECUTION_ID_JSON_PATH = "$..id";
+
+    private final JiraFacade jiraFacade;
+    private final JiraClient client;
+    private final ZephyrConfiguration zephyrConfiguration;
+
+    public ZephyrFacade(JiraFacade jiraFacade, JiraClient client, ZephyrConfiguration zephyrConfiguration)
+    {
+        this.jiraFacade = jiraFacade;
+        this.client = client;
+        this.zephyrConfiguration = zephyrConfiguration;
+    }
+
+    @Override
+    public Integer createExecution(String execution) throws IOException
+    {
+        String responseBody = client.executePost(CREATE_EXECUTION_ENDPOINT, execution);
+        List<Integer> executionId = JsonPathUtils.getData(responseBody, EXECUTION_ID_JSON_PATH);
+        return executionId.get(0);
+    }
+
+    @Override
+    public void updateExecutionStatus(int executionId, String executionBody) throws IOException
+    {
+        client.executePut(String.format(UPDATE_EXECUTION_STATUS_ENDPOINT, executionId), executionBody);
+    }
+
+    @Override
+    public ZephyrConfiguration prepareConfiguration() throws IOException
+    {
+        notBlank(zephyrConfiguration.getProjectKey(), "Property 'zephyr.project-key=' should not be empty");
+        notBlank(zephyrConfiguration.getVersionName(), "Property 'zephyr.version-name=' should not be empty");
+        notBlank(zephyrConfiguration.getCycleName(), "Property 'zephyr.cycle-name=' should not be empty");
+        notBlank(zephyrConfiguration.getFolderName(), "Property 'zephyr.folder-name=' should not be empty");
+
+        Project project = jiraFacade.getProject(zephyrConfiguration.getProjectKey());
+        String projectId = project.getId();
+        zephyrConfiguration.setProjectId(projectId);
```

Comment:
at some point it would be needed to split `ZephyrConfiguration ` into 2 classes: one should be immutable and should provide the user properties, another one will collect dynamic configuration